mender-convert

```diff
@@ -266,9 +263,21 @@ do_raw_disk_image_create_partitions() {
 
   mkdir -p $output_dir && cd $output_dir
 
-  # In case of missing .sdimg name use the default format.
-  [ -z $mender_disk_image ] && mender_disk_image=$output_dir/mender_${device_type}.sdimg \
-                            || mender_disk_image=$output_dir/$mender_disk_image
+  # Make sure the user's given Mender image name has a correct extension.
+  # If Mender image name is not provided, then use following syntax:
+  # mender-<device_name>-<artifact_name>.sdimg
+  if [ -n "${mender_disk_image}" ]; then
+    mender_disk_basename=$(basename -- "$mender_disk_image")
+    if [[ $mender_disk_basename =~ \.sdimg$ ]]; then
+      mender_disk_image=$output_dir/$mender_disk_basename
+    else
+      mender_disk_filename="${mender_disk_basename%.*}"
```

Comment:
Have you tried running the script without "--mender-disk-image" argument? It seems that you only set certain variables if  that argument is passed but these variables are used unconditionally in other locations.

Reply:
The only result of that part of the script is that the variable _mender_disk_image_ is set. In other locations only that variable is used. 
_mender_disk_image_ is used in two ways:
- when _from-raw-disk-image_ is called, then its real content is build in _raw-disk-image-create-partitions_ and is propagated to other commands
- when one of the _Expert commands_ is invoked, then the path to Mender image is passed in CLI - _Expert command_ checks whether such a file exists

Reply:
I will trust you on that :)